.github/workflows/release.yml

```diff
@@ -0,0 +1,95 @@
+name: 'release'
+
+on:
+  push:
+    branches:
+    - 'main'
+
+jobs:
+  # build compiles the code and creates a pull request of the compiled result if
+  # there is a diff.
+  build:
+    runs-on: 'ubuntu-latest'
+    steps:
+    - uses: 'actions/checkout@v2'
+
+    - uses: 'actions/setup-node@v2'
+      with:
+        node-version: '12.x'
+
+    - name: 'npm build'
+      run: 'npm ci && npm run build'
+
+    - name: 'Commit'
+      run: |-
+        git config user.name "github-actions"
+        git config user.email "41898282+github-actions[bot]@users.noreply.github.com"
+        git add dist/
+        git commit -m "chore: build dist"
+
+    - name: 'Create pull request'
+      uses: 'peter-evans/create-pull-request@v3'
+      with:
+        committer: 'github-actions <41898282+github-actions[bot]@users.noreply.github.com>'
+        author: 'github-actions <41898282+github-actions[bot]@users.noreply.github.com>'
+        signoff: 'github-actions <41898282+github-actions[bot]@users.noreply.github.com>'
+        commit-message: 'Build dist'
+        title: 'chore: build dist'
+        body: 'Build compiled Typescript'
+        base: 'main'
+        branch: 'actions/build'
+        delete-branch: true
+
+  # create-pull-request creates a release pull request if there are any
+  # convential commit changes since the last release.
+  create-pull-request:
+    runs-on: 'ubuntu-latest'
+    steps:
+    - uses: 'google-github-actions/release-please-action@main'
+      with:
+        release-type: 'node'
+        bump-minor-pre-major: true
+        command: 'release-pr'
+
+  # release does a release on the merge of the release pull request. It also
+  # updates the floating tag alias for the major version.
+  release:
+    runs-on: 'ubuntu-latest'
+    steps:
+    - id: 'release'
+      uses: 'google-github-actions/release-please-action@main'
+      with:
+        release-type: 'node'
+        bump-minor-pre-major: true
+        command: 'github-release'
+
+    - name: 'Update floating tag'
+      if: '${{ steps.release.outputs.release_created }}'
```

Comment:
Should we make this workflow dispatch too since rerunning should be idempotent. IIUC based on our offline discussion, you were also looking into manual tag support?

Reply:
Done